src/index.ts

```diff
@@ -308,21 +326,30 @@ export class Probot {
   public start() {
     this.log.info(`Running Probot v${this.version}`);
     this.httpServer = this.server
-      .listen(this.options.port, () => {
-        if (this.options.webhookProxy) {
-          createWebhookProxy({
-            logger: this.log,
-            path: this.options.webhookPath,
-            port: this.options.port,
-            url: this.options.webhookProxy,
-          });
+      // FIXME If types are not forced compiler complains with:
+      // Argument of type 'number | undefined' is not assignable to parameter of type 'number' and
+      // Argument of type 'string | undefined' is not assignable to parameter of type 'string'
+      .listen(
+        Number(this.options.port),
+        this.options.bindAddress as string,
```

Comment:
I think we have to set default options within the Probot constructor, as it can be called directly, too. Setting defaults in `Probot.run` is not sufficient.

Reply:
Actually the problem seems to be related with the fact that `options.port` and `options.bindAddress` are currently defined as optional properties, removing the ending `?` from both in the type definition fixes it.

Reply:
Mmm, I'm not sure how to tackle this, I've put some comments in the latest patch, please advise

Reply:
I think what I would be is to not use `this.options`, but instead set `port` and `bindAddress` on `this.state`, where we can guarantee that it will be set